addons/mail/models/mail_thread.py

```diff
@@ -2881,8 +2890,19 @@ def _message_create(self, values_list):
             for x in ('from', 'to', 'cc'):
                 create_values.pop(x, None)
             create_values['partner_ids'] = [Command.link(pid) for pid in (create_values.get('partner_ids') or [])]
+            channel_ids = create_values.pop('channel_ids', None)
             create_values_list.append(create_values)
 
+            # Create Notification in mentioned channels
+            if channel_ids and (self._name != 'discuss.channel' or self.channel_type == 'channel'):
```

Comment:
And the whole code should probably be moved in discuss module override of mail.thread